src/Optimizer.ts

```diff
@@ -56,6 +62,20 @@ export class Optimizer {
    */
   private sortFunction = (a: ReportElement, b: ReportElement): number => { return (a.action.length - b.action.length || b.path.length - a.path.length);};
 
+  /**
+   * The complete list of getOptimizedDocument configuration options used to parse the optimized document.
+   * @typedef {Object} Rules
+   * @property {Boolean=} reuseComponents - whether to reuse components from `components` section or not. Defaults to `true`.
+   * @property {Boolean=} removeComponents - whether to remove un-used components from `components` section or not. Defaults to `true`.
+   * @property {Boolean=} moveToComponents - whether to move duplicated components to the `components` section or not. Defaults to `true`.
+   */
+
+  /**
+   * The complete list of getOptimizedDocument configuration options used to parse the optimized document.
```

Comment:
You have duplicated description for these two typedefs, please add appropriate :)

Reply:
@magicmatatjahu thanks for the quick review 😀 . removed both of them since the typedefs are pretty self-explanatory.